content/webapp/pages/events/index.tsx

```diff
@@ -232,30 +312,94 @@ const EventsPage: FunctionComponent<Props | NewProps> = props => {
+
+                    updateUrl(event.currentTarget);
+                  }}
+                >
+                  <SearchFilters
+                    linkResolver={params =>
+                      linkResolver({
+                        params,
+                        pathname: `${router.pathname}${isInPastListing ? '/past' : ''}`,
+                      })
+                    }
+                    searchFormId={EVENTS_LISTING_FORM_ID}
+                    changeHandler={() => {
+                      const form = document.getElementById(
+                        EVENTS_LISTING_FORM_ID
+                      );
+                      form &&
+                        form.dispatchEvent(
+                          new window.Event('submit', {
+                            cancelable: true,
+                            bubbles: true,
+                          })
+                        );
+                    }}
+                    filters={filters}
+                    hasNoResults={hasNoResults}
+                  />
+                </form>
+              </Space>
+            </ContaineredLayout>
+          )}
+          {hasNoResults ? (
+            <ContaineredLayout gridSizes={gridSize12()}>
+              <p>No {isInPastListing ? 'past ' : 'upcoming'} events found.</p>
+            </ContaineredLayout>
+          ) : (
+            <>
+              <ContaineredLayout gridSizes={gridSize12()}>
+                <PaginationWrapper $verticalSpacing="l">
+                  <span>
+                    {pluralize(
+                      events.totalResults,
+                      (isInPastListing ? 'past ' : '') + 'event'
+                    )}
+                  </span>
+                  {activeFiltersLabels.length > 0 && (
+                    <span className="visually-hidden">
+                      {' '}
+                      filtered with: {activeFiltersLabels.join(', ')}
+                    </span>
+                  )}
+
+                  <Pagination
+                    totalPages={events.totalPages}
+                    ariaLabel="Events top pagination"
+                    isHiddenMobile
+                  />
+                </PaginationWrapper>
+
+                <EventsSearchResults
+                  events={events.results}
+                  isInPastListing={isInPastListing}
+                />
+              </ContaineredLayout>
 
-          <ContaineredLayout gridSizes={gridSize12()}>
-            <PaginationWrapper $verticalSpacing="l" $alignRight>
-              <Pagination
-                totalPages={events.totalPages}
-                ariaLabel="Results pagination"
-              />
-            </PaginationWrapper>
-          </ContaineredLayout>
+              <ContaineredLayout gridSizes={gridSize12()}>
+                <PaginationWrapper $verticalSpacing="l" $alignRight>
+                  <Pagination
+                    totalPages={events.totalPages}
+                    ariaLabel="Events bottom pagination"
```

Comment:
A bit of a nit, but I'm not sure the position of the pagination on the page is of use/interest to people using assistive tech – I think just 'Events pagination' (or indeed 'Results pagination') might be better?

Reply:
I thought it might be confusing to have the two, happy to change it back!